### PR TITLE
[TC-138] Gem documentation

### PR DESCRIPTION
**Pull request by tiagoamaro**
This PR creates documentation for the `EventSourcing` gem, describing this framework's concepts and how to add it into your project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,6 @@
-# Event::Source
+# EventSourcing
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/event/source`. To experiment with that code, run `bin/console` for an interactive prompt.
-
-TODO: Delete this and the text above, and describe your gem
+Simple PORO Event Sourcing framework, with Rails support.
```
Comment (nullobject) [inline README.md]: I'm not sure that most people will know what "PORO" means. You could probably substitute "Ruby" everywhere you say "PORO".

This is the introduction section, you should probably also briefly describe what an "event sourcing framework" is and why you might need one. Just a paragraph or two.

File: README.md [hunk 2]
```diff
@@ -19,25 +17,78 @@ And then execute:
...
 ## License
 
 The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).
 
 ## Code of Conduct
 
-Everyone interacting in the Event::Source project’s codebases, issue trackers, chat rooms and mailing lists is expected to follow the [code of conduct](https://github.com/conversation/event-sourcing/blob/master/CODE_OF_CONDUCT.md).
+Everyone interacting in the The Conversation project’s codebases, issue trackers, chat rooms and mailing lists is
+
+expected to follow the [code of conduct](https://github.com/conversation/event-sourcing/blob/master/CODE_OF_CONDUCT.md).
```
Comment (nullobject) [inline README.md]: No need for a new paragraph.

File: README.md [hunk 3]
```diff
@@ -19,25 +17,78 @@ And then execute:
...
+```ruby
+gem "event-sourcing", require: "event_sourcing/rails"
+```
+
+Further installation steps and a sample implementation can be found at the [Rails' README file](examples/rails_app/README.md).
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+- System requirements
```
Comment (nullobject) [inline README.md]: I would move everything under the "Development" section to "Contributing". Keep things simple.

File: README.md [hunk 4]
```diff
@@ -19,25 +17,78 @@ And then execute:
...
-Bug reports and pull requests are welcome on GitHub at https://github.com/conversation/event-sourcing. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [Contributor Covenant](http://contributor-covenant.org) code of conduct.
+Bug reports and pull requests are welcome on GitHub at https://github.com/conversation/event-sourcing.
+
+This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere
+
+to the [Contributor Covenant](http://contributor-covenant.org) code of conduct.
 
 ## License
 
 The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).
```
Comment (nullobject) [inline README.md]: You should link to the project license file here.

File: README.md [hunk 5]
```diff
@@ -19,25 +17,78 @@ And then execute:
...
+    - "Applying" an event is also known as "Calculator" on other event sourcing literature
+- **Commands**: validates and persists events
+- **Reactors**: trigger side effects as events happen
+- **Event Dispatcher**: manage which reactors will be executed after an event happens
+
+While state changing actions obey the rationale:
+
+- Aggregate state changes should be executed through a Command
+- Events are immutable, as they represent series of changes to an Aggregate
+- Reactors may be triggered after an Event occur
```
Comment (nullobject) [inline README.md]: "occurs"

**Comment by tiagoamaro** [comment]
@nullobject, probably not there yet, but made several changes to the README as you requested 👏 (https://github.com/conversation/event-sourcing/pull/5/commits/67a4fe9965fee2ca78cc9fb9abfd26fe0cb4fe4d).

**Comment by tiagoamaro** [comment]
Awesome @nullobject! Will rebase ASAP.

**Pull request merged**